src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -225,6 +231,50 @@ public void updateCheesesStatus(Set<CheeseId> cheesesAssigned) {
         cheeses.updateCheesesStatus(cheesesAssigned);
     }
 
+    /**
+     * Checks whether the address book's current state is valid
+     * Includes checking dependencies between models
+     */
+    public void checkAddressBook() {
+        Set<CustomerId> customerIdSet = customers.asUnmodifiableObservableList().stream()
+            .map(x -> x.getId()).collect(Collectors.toSet());
+        Map<CheeseId, Cheese> cheeseIdMap = cheeses.asUnmodifiableObservableList().stream()
+            .collect(Collectors.toMap(x -> x.getCheeseId(), x -> x));
+
+        HashSet<CheeseId> orderCheeseIdSet = new HashSet<>();
+        // Check that each order is valid on the system level
+        for (Order order : orders.asUnmodifiableObservableList()) {
+            CustomerId customerId = order.getCustomerId();
+            checkArgument(customerIdSet.contains(customerId), "Order "
+                + order.getOrderId() + "'s customer ID does not exist.");
+
+            CheeseType expectedCheeseType = order.getCheeseType();
+
+            Set<CheeseId> cheeseIds = order.getCheeses();
+            OrderId orderId = order.getOrderId();
+
+            if (cheeseIds.size() > 0) {
+                for (CheeseId cheeseId : cheeseIds) {
+                    checkArgument(cheeseIdMap.containsKey(cheeseId),
+                        "Order " + orderId + "'s cheese ID does not exist.");
+
```

Comment:
I think it's okay if the cheese ID does not currently exist?

Consider the following scenario:
- User adds a new cheese (CheeseID = 1, CheeseType = Gouda, Quantity = 1)
- User adds a new order (OrderID = 1, CheeseType = Gouda, Quantity = 1)
- User completes Order 1 with Cheese 1.
- User deletes Cheese 1 (we currently allow this to happen without changing the order to which the cheese was assigned).
- User close the app.
- User reopens the app and runs into an IllegalArgumentException because CheeseID 1 no longer exist.

Reply:
I think this might cause some trouble on our side cause imagine if Order shows all the cheeses that are assigned to it (and also cause order needs to check that the cheese is of the same type).

I think, we can possibly disallow cheese deletion if the cheese has been assigned? (cause I think delete cascading will be bad for this)

Reply:
I think disallowing cheese deletion after it has been assigned sounds good - it will be just like IRL where you can't throw away a cheese after assigning it to an order, for example.